frontend/src/app/components/crm/leads/leads.component.ts

```diff
@@ -99,27 +99,27 @@ export class LeadsComponent implements AfterViewInit {
        * @param data The whole data we have in the JSON.
        * @param filter The value that the user searches for.
        */
-      this.dataSource.filterPredicate = (data: any, filter: string): boolean  => {
+      this.dataSource.filterPredicate = (filterPredicateData: any, filterPredicateFilter: string): boolean  => {
         const cleanString = (str: string): string => str.trim().toLowerCase();
         const hasFilter = (data: any, filter: string): boolean => {
           // traverse through JSON's tree like structure
-          if(typeof data === 'object') {
-            for(const key of Object.keys(data)) {
-              if(hasFilter(data[key], filter)) {
+          if (typeof data === 'object') {
+            for (const key of Object.keys(data)) {
+              if (hasFilter(data[key], filter)) {
                 return true;
               }
             }
           } else {
             // if you hit a key-value pair where the value is
             // a primitve type compare and return only if filter found
             const value = cleanString(_.toString(data));
-            if(value.indexOf(filter) !== -1) {
+            if (value.indexOf(filter) !== -1) {
```

Comment:
Shouldn't data be renamed to filterPredicateData and filter be renamed to filterPredicateFilter in the method?

Reply:
Nope. I renamed to differentiate between `filterPredicate`'s parameters and `hasFilter`'s parameters. The Typescript linter actually caught this. https://palantir.github.io/tslint/rules/no-shadowed-variable/